File: src/main/java/com/dogGetDrunk/meetjyou/preference/UserPreference.java
package com.dogGetDrunk.meetjyou.preference;

import com.dogGetDrunk.meetjyou.user.User;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "UserPreference")
public class UserPreference {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private User user;

    @ManyToOne
    private Preference preference;

    public UserPreference(User user, Preference preference) {
        this.user = user;
        this.preference = preference;
    }
}

File: src/main/java/com/dogGetDrunk/meetjyou/preference/UserPreference.kt
package com.dogGetDrunk.meetjyou.preference

import com.dogGetDrunk.meetjyou.user.User
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne

@Entity
class UserPreference(
    @ManyToOne
    var user: User,

    @ManyToOne
    var preference: Preference
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0
}
